applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py

```diff
@@ -1,104 +1,124 @@
 
         ## Construct the linear solver
         import linear_solver_factory
         self.linear_solver = linear_solver_factory.ConstructSolver(self.settings["linear_solver_settings"])
 
         # Initialize shallow water variables utility
-        self.ShallowVariableUtils = KratosShallow.ShallowWaterVariablesUtility(self.model_part, self.settings["dry_height"].GetDouble())
+        self.ShallowVariableUtils = Shallow.ShallowWaterVariablesUtility(self.main_model_part, self.settings["dry_height"].GetDouble())
 
     def AddVariables(self):
-        # Basic variables
-        self.model_part.AddNodalSolutionStepVariable(KratosShallow.HEIGHT)
-        self.model_part.AddNodalSolutionStepVariable(KratosMultiphysics.VELOCITY)
+        # Primitive variables
+        self.main_model_part.AddNodalSolutionStepVariable(Shallow.HEIGHT)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.VELOCITY)
         # Physic problem parameters
-        self.model_part.AddNodalSolutionStepVariable(KratosShallow.FREE_SURFACE_ELEVATION)
-        self.model_part.AddNodalSolutionStepVariable(KratosMultiphysics.GRAVITY)
-        self.model_part.AddNodalSolutionStepVariable(KratosShallow.BATHYMETRY)
-        self.model_part.AddNodalSolutionStepVariable(KratosShallow.MANNING)
-        self.model_part.AddNodalSolutionStepVariable(KratosShallow.RAIN)
+        self.main_model_part.AddNodalSolutionStepVariable(Shallow.FREE_SURFACE_ELEVATION)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.GRAVITY)
+        self.main_model_part.AddNodalSolutionStepVariable(Shallow.BATHYMETRY)
+        self.main_model_part.AddNodalSolutionStepVariable(Shallow.MANNING)
+        self.main_model_part.AddNodalSolutionStepVariable(Shallow.RAIN)
         # Auxiliary variables
-        self.model_part.AddNodalSolutionStepVariable(KratosMultiphysics.IS_STRUCTURE)
-        self.model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NORMAL)
-        self.model_part.AddNodalSolutionStepVariable(KratosMultiphysics.MESH_VELOCITY)
-        self.model_part.AddNodalSolutionStepVariable(KratosMultiphysics.DISPLACEMENT)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.IS_STRUCTURE)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.NORMAL)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.MESH_VELOCITY)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.DISPLACEMENT)
 
     def AddDofs(self):
         raise Exception("Calling the base class instead of the derived one")
 
+    def ImportModelPart(self):
+        # we can use the default implementation in the base class
+        self._ImportModelPart(self.main_model_part,self.settings["model_import_settings"])
+
+    def PrepareModelPart(self):
+        # Defining the variables
+        gravity = self.settings["gravity"].GetDouble()
```

Comment:
don't you prefer to set the gravity with a process?

Reply:
there is something strange

Reply:
sorry, it is the wrong branch